subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/SetValueSnapshot.java

```diff
@@ -67,4 +72,17 @@ public boolean equals(Object obj) {
     public int hashCode() {
         return elements.hashCode();
     }
+
+    @Override
+    public Set isolate() {
+        Set set = new HashSet();
```

Comment:
⭕️ I think we have to use LinkedHashSet here, as the original might have been linked as well.